libs/shared/form-fields/src/lib/FileUploadController/index.tsx

```diff
@@ -0,0 +1,251 @@
+import React, { FC, useState, useReducer, useEffect } from 'react'
+import { getValueViaPath, Application } from '@island.is/application/core'
+import {
+  InputFileUpload,
+  UploadFile,
+  fileToObject,
+} from '@island.is/island-ui/core'
+import { useFormContext, Controller } from 'react-hook-form'
+import { useMutation } from '@apollo/client'
+
+import { uploadFileToS3 } from './utils'
+import { Action, ActionTypes } from './types'
+
+import {
+  CREATE_UPLOAD_URL,
+  ADD_ATTACHMENT,
+  DELETE_ATTACHMENT,
+} from '@island.is/application/graphql'
+import { useLocale } from '@island.is/localization'
+
+type UploadFileAnswer = {
+  name: string
+  key?: string
+  url?: string
+}
+
+// Transform an uploaded file to an form answer.
+const transformToAnswer = (f: UploadFile): UploadFileAnswer => {
```

Comment:
You could destructurate to make it easier for the return

```suggestion
const transformToAnswer = (({ name, key, url }): UploadFile): UploadFileAnswer => {
  return { name, key, url }
```

Reply:
Nice, agree, switched now.